llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp

```diff
@@ -229,13 +229,13 @@ class SchedGroup {
   SchedGroup(SchedGroupMask SGMask, std::optional<unsigned> MaxSize,
              ScheduleDAGInstrs *DAG, const SIInstrInfo *TII)
       : SGMask(SGMask), MaxSize(MaxSize), DAG(DAG), TII(TII) {
-    SGID = NumSchedGroups++;
+    SGID = __atomic_fetch_add(&NumSchedGroups, 1, __ATOMIC_SEQ_CST);
```

Comment:
Nothing should be atomic here

Reply:
Why not? (Maybe I misunderstood something)

Reply:
There is no multithread compilation or shared state. The thread safety risk is from different LLVMContexts in different threads, not within the single compilation context. Multiple sequential compiles in the same process are also broken by static variables

Reply:
alright, thanks